crates/services/upgradable-executor/src/executor.rs

```diff
@@ -1264,6 +1272,51 @@ mod test {
         }
     }
 
+    /// wrapper type to ensure correct ordering of version strings
+    #[derive(Debug, Eq, PartialEq, Clone)]
+    struct VersionKey(String);
+
+    impl VersionKey {
+        fn new(version: String) -> Self {
+            VersionKey(version)
+        }
+
+        fn as_str(&self) -> &str {
+            &self.0
+        }
+    }
+
+    impl Ord for VersionKey {
+        fn cmp(&self, other: &Self) -> Ordering {
+            let self_parts: Vec<u32> = self
+                .0
+                .split('.')
+                .map(|s| s.parse::<u32>().unwrap_or(0))
+                .collect();
+
+            let other_parts: Vec<u32> = other
+                .0
+                .split('.')
+                .map(|s| s.parse::<u32>().unwrap_or(0))
```

Comment:
Is this default of `0` what the user would expect? Idk if as a user I'd expect '0' or `u32::MAX` or what.

Would it be possible to just impl `PartialOrd` and not `Ord`?

Reply:
NVM. Only under `#[cfg(test)]`